deployment/deploy/2060-deploy-VestingLogic.js

```diff
@@ -0,0 +1,85 @@
+const path = require("path");
+const { getContractNameFromScriptFileName } = require("../helpers/utils");
+const col = require("cli-color");
+const { sendWithMultisig } = require("../helpers/helpers");
+const func = async function (hre) {
+    const {
+        deployments: { deploy, get, log },
+        getNamedAccounts,
+        ethers,
+    } = hre;
+    const { deployer } = await getNamedAccounts(); //await ethers.getSigners(); //
+    let totalGas = ethers.BigNumber.from(0);
+
+    // Deploy vestingLogic //
+    log(col.bgYellow("Deploying VestingLogic..."));
+    await deploy("VestingLogic", {
+        from: deployer,
+        args: [],
+        log: true,
+        contract: "VestingLogic",
+    });
+
+    const vestingLogicDeployment = await get("VestingLogic");
+
+    // Deploy vestingFactory //
+    log(col.bgYellow("Deploying VestingFactory..."));
+    await deploy("VestingFactory", {
+        from: deployer,
+        args: [vestingLogicDeployment.address],
+        log: true,
+        contract: "VestingFactory",
+    });
+
+    const vestingFactoryDeployment = await get("VestingFactory");
+    const vestingRegistry = await ethers.getContract("VestingRegistry");
+    const staking = await ethers.getContract("Staking");
+
+    /** VestingRegistry still owned by Exchequer for Mainnet & tetstnet */
```

Comment:
pls consider differentiating upgrades based on actual ownership, not network and special cases

Reply:
Apparently the VestingRegistry owned by governance, so i updated the deployment script in this Updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/529/commits/eaf61228f6677387d6ee5b37b08ed83a61ceb269)

Reply:
why isn't this one addressed?

Reply:
actually, can be extreacted to a separate ticket - we need to refactor all the deployment scripts

Reply:
resolved